syftlib/src/main/java/org/openmined/syft/utilities/NetworkUtils.kt

```diff
@@ -0,0 +1,141 @@
+package org.openmined.syft.utilities
+
+import android.util.Log
+import io.reactivex.Single
+import okhttp3.ResponseBody
+import java.io.File
+import java.io.InputStream
+import java.lang.Integer.min
+
+
+private const val TAG = "NetworkUtils"
+private const val SPEED_BUFFER_WINDOW = 20
+private const val SPEED_MULTIPLICATION_FACTOR = 10
+private const val MB = 1024 * 1024
+private const val MAX_SPEED_TESTING_BYTES = MB * 8
+
+object NetworkUtils {
```

Comment:
Instead of `NetworkUtils` why not having a `RemoteDataSource` and a repository to work with it?

Reply:
This class would turn into a datasource except for the function downloadSpeedtest. T doesn't fit well into datasource construct acc to me. Don't know where to keep it. Removing this The class can become file datasource. The repository will still be syftjob and syft classes